### PR TITLE
docs(api): add `definePageMeta` util

### PR DESCRIPTION
**Pull request by Krutie**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Added API documentation for `definePageMeta` utility function

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 1]
```diff
@@ -1,7 +1,131 @@
...
+
+Set `key` value when you need more control over when the `<NuxtPage>` component is re-rendered.
+
+- **layout** - `boolean | string`  
+
+Set a static or dynamic name of the layout for each route. This can be set to `false` as well in case the default layout needs to be disabled.
+
+- **middleware**: `string | function`
+
+Define anonymous or named middleware directly within `definePageMeta` under `middleware` key. Learn more about route middleware.
```
Comment (danielroe) [inline docs/content/3.api/3.utils/define-page-meta.md:37]:
```suggestion
Define anonymous or named middleware directly within `definePageMeta` under `middleware` key. [Learn more about route middleware](/docs/directory-structure/middleware/).
```

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 2]
```diff
@@ -1,7 +1,131 @@
...
+
+Set name of the transition to apply for current layout. You can also set this value to `false` to disable the layout transition.
+(Learn more about [TransitionProps](https://github.com/vuejs/vue/blob/main/src/platforms/web/runtime/components/transition.ts)).
+
+- **pageTransition**: `boolean | TransitionProps`
+
+Set name of the transition to apply for current page. You can also set this value to `false` to disable the page transition.
+
+- **alias**: `string | string[]`
+Set alias to access the same page from different paths. Learn more about alias [alias](https://router.vuejs.org/guide/essentials/redirect-and-alias.html#alias)
```
Comment (danielroe) [inline docs/content/3.api/3.utils/define-page-meta.md:49]:
```suggestion

Set alias to access the same page from different paths. Learn more about [alias](https://router.vuejs.org/guide/essentials/redirect-and-alias.html#alias)
```

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 3]
```diff
@@ -1,7 +1,131 @@
...
+
+- **keepalive**: `boolean | KeepAliveProps`
+
+Set to `true` when you want to preserve page state across route changes.
+(Learn more about [KeepAlive](https://vuejs.org/api/built-in-components.html#keepalive))
```
Comment (danielroe) [inline docs/content/3.api/3.utils/define-page-meta.md:54]:
```suggestion
Set to `true` when you want to preserve page state across route changes. Learn more about [KeepAlive](https://vuejs.org/api/built-in-components.html#keepalive).
```

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 4]
```diff
@@ -1,7 +1,131 @@
...
+
+Apart from above values, you can set **custom** values in `definePageMeta` as well depending on your use-case. This data can then be accessed throughout your Nuxt application using `route.meta` object as shown below.
```
Comment (danielroe) [inline docs/content/3.api/3.utils/define-page-meta.md:56]:
```suggestion
Apart from the above values, you can also set **custom** values in `definePageMeta` as well, depending on your use-case. This data can then be accessed throughout your Nuxt application using the `route.meta` object as shown below.
```

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 5]
```diff
@@ -1,7 +1,131 @@
...
+  const route = useRoute()
+  console.log(route.meta.title)
+</script>
+```
+
+## Examples
+
+### Basic usage
+
+The example below shows how `key` can be a function that returns a value, while `pageType` is a custom property and `keepAlive` property makes sure that the `modal` component is not cached when you change pages.
```
Comment (danielroe) [inline docs/content/3.api/3.utils/define-page-meta.md:69]:
```suggestion
The example below shows how `key` can be a function that returns a value, while `pageType` is a custom property and `keepalive` property makes sure that the `modal` component is not cached when you change pages.
```

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 6]
```diff
@@ -1,7 +1,131 @@
...
+      exclude: ['modal']
+    },
+
+    // custom property: pageType
+    pageType: "Checkout"
+  })
+</script>
+```
+
+### Define m**iddleware**
```
Comment (danielroe) [inline docs/content/3.api/3.utils/define-page-meta.md:87]:
```suggestion
### Define middleware
```

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 7]
```diff
@@ -1,7 +1,131 @@
...
+        if (!auth.value.authenticated) {
+            return navigateTo('/login')
+        }
+       return navigateTo('/checkout')
+      }
+    ],
+
+    // string
+    // middleware: ["auth"]
+    // or middleware: "auth"
```
Comment (danielroe) [inline docs/content/3.api/3.utils/define-page-meta.md:108]:
```suggestion
    // middleware: ['auth']
    // or middleware: 'auth'
```

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 8]
```diff
@@ -1,7 +1,131 @@
...
+```
+
+### Define layout
+
+You can set the `layout` as a `string` value that matches the layout file located in `layouts/` directory. You can disable the layout by setting the `layout` to `false`.
+
+```vue [pages/some-page.vue]
+<script setup>
+  function applyCustomLayout () {
+    route.meta.layout = "custom"
```
Comment (danielroe) [inline docs/content/3.api/3.utils/define-page-meta.md:120]:
```suggestion
    route.meta.layout = 'custom'
```

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 9]
```diff
@@ -1,7 +1,131 @@
...
+  }
+
+  definePageMeta({
+    layout: false
+    // or layout: "custom"
```
Comment (danielroe) [inline docs/content/3.api/3.utils/define-page-meta.md:125]:
```suggestion
    // or layout: 'custom'
```

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 10]
```diff
@@ -1,7 +1,131 @@
...
+  definePageMeta({
+    title: 'Checkout home page',
+    key: route => route.fullPath,
+
+    keepalive: {
+      exclude: ['modal']
+    },
+
+    // custom property: pageType
+    pageType: "Checkout"
```
Comment (danielroe) [inline docs/content/3.api/3.utils/define-page-meta.md:82]:
```suggestion
    pageType: 'Checkout'
```

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 11]
```diff
@@ -1,7 +1,131 @@
...
+```
+
+> `definePageMeta` works in both `<script setup>` and `<script>`.
+
+## Properties
+
+You can set following properties through `definePageMeta` .
+
+- **title** - `string`
+Set custom title for each route of your application
```
Comment (danielroe) [inline docs/content/3.api/3.utils/define-page-meta.md:25]:
```suggestion

Set custom title for each route of your application
```

File: docs/content/3.api/3.utils/define-page-meta.md [hunk 12]
```diff
@@ -1,7 +1,132 @@
...
+
+- **key** - `string | function`
+
+Set `key` value when you need more control over when the `<NuxtPage>` component is re-rendered.
+
+- **layout** - `boolean | string`  
+
+Set a static or dynamic name of the layout for each route. This can be set to `false` as well in case the default layout needs to be disabled.
+
+- **middleware**: `string | function`
```
Comment (pi0) [inline docs/content/3.api/3.utils/define-page-meta.md:36]: I think this can be an array too

Comment (DamianGlowala) [reply docs/content/3.api/3.utils/define-page-meta.md:36]: Good catch:

`PageMeta.middleware?: string | NavigationGuard | (string | NavigationGuard)[] | undefined`

**Review by DamianGlowala — APPROVED** [review]

**Review by pi0 — APPROVED** [review]

**Pull request merged**